quest/molecule.py

```diff
@@ -0,0 +1,49 @@
+import psi4
+
+class Molecule(object):
+    '''
+    A molecule class that is used to store basic parameters including
+    psi4 geometry, basis set, number of electrons, and number of doubly
+    occupied orbitals.
+    '''
+
+    def __init__(self, mol=None, bas=None):
+        '''
+        Initialize molecule with geometry mol and basis
+        set bas. Calculate number of electrons nel and
+        number of doubly occupied orbitals ndocc.
+        '''
+        self.mol = mol
+        self.bas = bas
+
+        # Calculate the number of doubly occupied orbitals and the number of electrons
+        self.nel = sum(self.molecule.Z(n) for n in range(self.molecule.natom()))
+        self.nel -= self.molecule.molecular_charge()
+        
+        if not (self.nel / 2.0).is_integer():
+            raise ValueError("Molecule must have an even number of electrons to perform RHF.")
+        self.ndocc = int(self.nel / 2.0)
```

Comment:
You may need to call `set_geometry` and `set_basis` if these are not `None`.

`if mol is not None: set_geometry(mol)`.